evm_arithmetization/src/cpu/kernel/interpreter.rs

```diff
@@ -524,6 +548,145 @@ impl<F: RichField> State<F> for Interpreter<F> {
         }
     }
 
+    fn check_against_struct_logs_before_op(
+        &mut self,
+        opcode: u8,
+        is_user_mode: bool,
+    ) -> Result<(), ProgramError> {
+        if let Some(struct_logs) = &self.struct_logs
+            && is_user_mode
+            && log_enabled!(log::Level::Debug)
```

Comment:
We shouldn't have to require Debug (which is extremely noisy within `evm_arithmetization` crate). Instead we should have a possibility to toggle on/off on-demand this feature of checking consistency

Reply:
Hmm... So can we just use the `get_struct_logs` flag (on zero-bin an trace_decoder) as a way to determine whether we should run the checks (from my understanding, Einar's PR would always introduce some structlogs, so that flag should still be usable even after that's merged, right?), and then we can just check whether `self.struct_logs` is some to determine whether we should run the checks? Or do you think it would be better to have an additional feature for it specifically?

Reply:
This is different. We get the structlogs in Einar's PR for performance, this hits performances. We don't want to have it run in regular runs, but only when some debugging is needed.
If you were asking about reusing the struct logs then yes. We don't need to fetch them twice

Reply:
Yes, but my question was: in this PR, we add a flag `get_struct_logs`, which I think could be used to set the new `struct_logs` in the Interpreter to either `Some(..)` or `None` (here, it's used to fetch the struct_logs, but once Einar's PR is merged, the prupose could be slightly changed). And thus, since we only have `Some` logs in the interpreter if `get_struct_logs` is on, then we can just use that as a way to determine whether we should run the checks?

Reply:
Hmm, I don't think I understand. Once the JD PR lands, we will _always_ have the struct logs. It doesn't mean we always want to run this debugging check, even if the struct logs are available.